WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -2328,6 +2325,53 @@ public void onMediaUploadCancelClicked(String localMediaId) {
     }
 
     @Override
+    public void onMediaDeleted(String localMediaId) {
+        if (!TextUtils.isEmpty(localMediaId)) {
+            // passing false here as we need to keep the media item in case the user wants to undo
+            cancelMediaUpload(StringUtils.stringToInt(localMediaId), false);
+        }
+    }
+
+    private void cancelMediaUpload(int localMediaId, boolean delete) {
+        MediaModel mediaModel = mMediaStore.getMediaWithLocalId(Integer.valueOf(localMediaId));
+        if (mediaModel != null) {
+            CancelMediaPayload payload = new CancelMediaPayload(mSite, mediaModel, delete);
+            mDispatcher.dispatch(MediaActionBuilder.newCancelMediaUploadAction(payload));
+        }
+    }
+
+    @Override
+    public void onUndoMediaCheck(final String undoedContent) {
```

Comment:
I get your point, but the thing is Aztec doesn't know how to make diffs, it just saves the whole string each time an action is triggered, replacing the editor's content with what was there before / after. The diff in this PR is done by inferring that in an older content (`undoedContent`) all we need to be interested in are the media items marked with the `UPLOAD` class, and that is not known to Aztec itself but rather something that belongs to AztecEditorFragmnet / WPAndroid app.
I think it would make things a bit more complicated if we gave Aztec this kind of capability to interpret its content any further than this.